src/linalg.jl

```diff
@@ -92,8 +81,115 @@ end
     @invcheckoff for multivec_row in 1:size(X.parent, 2)
         for col in 1:size(A, 2)
             @inbounds for k in nzrange(A, col)
-                C[multivec_row, col] += X.parent[A.rowval[k], multivec_row]*A.nzval[k]*α
+                @routine begin
+                    anc ← zero(eltype(A))
+                    anc += X.parent[A.rowval[k], multivec_row]' * A.nzval[k]
+                end
+                C[multivec_row, col] += anc * α
+                ~@routine
+            end
+        end
+    end
+end
+
+@i function imul!(C::StridedVecOrMat, X::DenseMatrixUnion, xA::Adjoint{<:Any,<:AbstractSparseMatrix}, α::Number, β::Number)
+    @safe size(X, 2) == size(xA.parent, 2) || throw(DimensionMismatch())
+    @safe size(X, 1) == size(C, 1) || throw(DimensionMismatch())
+    @safe size(xA.parent, 1) == size(C, 2) || throw(DimensionMismatch())
+    if (β != 1, ~)
+        @safe error("only β = 1 is supported, got β = $(β).")
+    end
+    @invcheckoff for col in 1:size(xA.parent, 2)
+        @inbounds for k in nzrange(xA.parent, col)
+                        @routine begin
+                            anc1 ← zero(eltype(xA.parent))
+                            anc1 += (xA.parent.nzval[k])'*α
+                        end
+            @simd for multivec_row in 1:size(X,1)
+                C[multivec_row, xA.parent.rowval[k]] += X[multivec_row, col] * anc1
+            end
```

Comment:
I didn't run it so just a question. Does this `@simd` annotation work here? Say, does it bring performance differences?

Reply:
Is allowed here, whether it brings speed up requires benchmarking.